GVFS/GVFS.Common/Prefetch/BlobPrefetcher.cs

```diff
@@ -229,6 +233,17 @@ public virtual void Prefetch(string branchOrCommit, bool isBranch)
                 }
             }
 
+            // TODO: make this a separate public method so that the caller can exit even earlier. Right now we're still wasting time
+            //       on auth and config even when we know there will be nothing to do
+            // TODO: always log the details from the last prefetch, to enable debugging of unexpected behaviors
+            if (this.IsNoopPrefetch(commitToFetch, hydrateFilesAfterDownload))
+            {
+                // TODO: prevent the CLI output from saying 0 matched/already cached

```

Comment:
Note: this is the cause of the one test failure:

```
2018-11-27T18:52:57.9017551Z Stats:
2018-11-27T18:52:57.9017646Z   Matched blobs:    0
2018-11-27T18:52:57.9017724Z   Already cached:   0
2018-11-27T18:52:57.9017808Z   Downloaded:       0
```

Reply:
(Which means the noop behavior is working. Just need to update the output, and also fix up the tests to expect or prevent the noop)